gen/suites/vm_violations/main.go

```diff
@@ -189,4 +190,42 @@ func main() {
 			MessageFunc: actorResolutionSecpExistant,
 		},
 	)
+
+	valPfx := "vm_violations/state_mutation/"
+
+	g.Group("state_mutation",
+		&VectorDef{
+			Metadata: &Metadata{
+				ID:      "in-transaction",
+				Version: "v1",
+				Desc:    "test an actor can mutate state within a transaction",
+			},
+			Selector:    map[string]string{"chaos_actor": "true"},
+			MessageFunc: mutateState(valPfx+"in-transaction", chaos.MutateInTransaction, exitcode.Ok),
+		},
+		&VectorDef{
+			Metadata: &Metadata{
+				ID:      "readonly",
+				Version: "v1",
+				Desc:    "test an actor cannot ILLEGALLY mutate readonly state",
+				Comment: "should abort with SysErrorIllegalActor",
```

Comment:
Hum, what does it abort with right now?

Reply:
Should probably accompany this comment and the one below with a link to a GitHub issue on Lotus.

Reply:
Current abort code is `Ok`:
![Screenshot 2020-09-04 at 12 18 06](https://user-images.githubusercontent.com/152863/92233740-b82b8e00-eea8-11ea-8df0-2f94c0367928.png)

Issue in Lotus:
https://github.com/filecoin-project/lotus/issues/3545

Reply:
Uh-oh. [I feared this would be the case](https://github.com/filecoin-project/test-vectors/issues/56#issuecomment-677754473), because I didn't recall seeing any code that tracked invalid writes. Definitely a correctness bug.